prio/rand.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2018, Henry Corrigan-Gibbs
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. 
+ */
+
+#include <limits.h>
+#include <nss/nss.h>
+#include <nss/pk11pub.h>
+#include <nspr/prinit.h>
+#include <mprio.h>
+
+#include "debug.h"
+#include "rand.h"
+#include "util.h"
+
+#define CHUNK_SIZE 8192
+
+SECStatus
+rand_init (void)
+{
+  SECStatus rv = NSS_NoDB_Init (NULL);
```

Comment:
Is this the first time you need NSS? How do you ensure that you initialise this only once?

Reply:
Yes, this is the first time that we need NSS. The `Prio_init()` function calls this one when the library is initialized.

To make sure that NSS only gets initialized once, I will wrap the `NSS_NoDB_Init` call with a check to `NSS_IsInitialized`.